File: valid_parentheses.py
class Solution:
    def isValid(self,s):
        lst=[]
        dic={'(':')','{':'}','[':']'}
        for alph in s:
            if len(lst)==0:
                lst.append(alph)
            else:
                if alph==dic.get(lst[-1]):
                    lst.pop()
                else:
                    lst.append(alph)
        if len(lst)==0:
            return True
        return False
if __name__=='__main__':
    sol=Solution()
    print sol.isValid('([)]')

File: Algorithm/CPP/valid_parentheses.cpp
# include <stdio.h>
# include <iostream>
# include <vector>
# include <string>
# include <cctype>
# include <stack>
# include <algorithm>
# include<time.h>
using namespace std;


class Solution {
public:
	bool isValid(string s) {
		stack<char> Stack;
		for (auto chr : s)
		{
			if (chr == '(' || chr == '[' || chr== '{')
				Stack.push(chr);
			if (chr == ')' || chr == ']' || chr == '}')
			{
				if (Stack.empty()) return false;
				if (chr == ')')
				{
					if (Stack.top() != '(') return false;
				}
				else if (chr == ']'){
					if (Stack.top() != '[') return false;
				}
				else {
					if (Stack.top() != '{') return false;
				}
				Stack.pop();
			}
				
		}
		return Stack.empty();
	}
};
int  main()
{
	Solution sol;

	//nums = { -1 };
	cout << sol.isValid("([)]");
	cin.get();
	return 0;
}

